### PR TITLE
[Eval] Refine CodeOwners

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by zhengfeiwang — DISMISSED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,7 +1,29 @@
...
+# 1. Folders
+/src/promptflow-tracing/ @microsoft/promptflow-execution
+
+/src/promptflow-eval/ @microsoft/pf-eval-team-contributor
+
+/src/promptflow-core/ @microsoft/promptflow-sdk
+/src/promptflow-devkit/ @microsoft/promptflow-sdk
+/src/promptflow-azure/ @microsoft/promptflow-sdk
+/src/promptflow-recording/ @microsoft/promptflow-sdk
+/src/promptflow-execution/ @microsoft/promptflow-sdk
```
Comment (wangchao1230) [inline .github/CODEOWNERS:16]: I think we don't have this folder? promptflow-execution

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
 
-/src/promptflow-tracing/promptflow/tracing/_start_trace.py @microsoft/promptflow-sdk
+* @microsoft/prompt-flow-approver
+
+# 1. Folders
+/src/promptflow-tracing/ @microsoft/promptflow-execution
+
+/src/promptflow-eval/ @microsoft/pf-eval-team-contributor
+
+/src/promptflow-core/ @microsoft/promptflow-sdk
```
Comment (wangchao1230) [inline .github/CODEOWNERS:12]: add one line for /src/promptflow?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,7 +1,29 @@
...
+# So put outside folders first, and then inside folders.
+# Don't change the sequence of the regions.
 
-/src/promptflow-tracing/promptflow/tracing/_start_trace.py @microsoft/promptflow-sdk
+* @microsoft/prompt-flow-approver
+
+# 1. Folders
+/src/promptflow-tracing/ @microsoft/promptflow-execution
+
+/src/promptflow-eval/ @microsoft/pf-eval-team-contributor
```
Comment (wangchao1230) [inline .github/CODEOWNERS:10]: make eval the last line

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -10,25 +32,14 @@
...
-/src/promptflow/tests/executor/ @microsoft/promptflow-execution
 
-/src/promptflow-core/promptflow/core/ @microsoft/promptflow-sdk
-/src/promptflow-devkit/promptflow/_cli/ @microsoft/promptflow-sdk
-/src/promptflow-devkit/promptflow/_sdk/ @microsoft/promptflow-sdk
-/src/promptflow-devkit/promptflow/_proxy/ @microsoft/promptflow-sdk
-/src/promptflow-devkit/promptflow/entities/ @microsoft/promptflow-sdk
-/src/promptflow-devkit/promptflow/operations/ @microsoft/promptflow-sdk
-/src/promptflow-azure/promptflow/azure/ @microsoft/promptflow-sdk
+# 3. Test folder extras
```
Comment (wangchao1230) [inline .github/CODEOWNERS:36]: shall we group 2, 3, 4 together by team ?

File: .github/CODEOWNERS (hunk 3 above)
Comment (wangchao1230) [reply .github/CODEOWNERS:10]: order: 

/src/promptflow
tracing
core
devkit
azure
recording
eval

File: .github/CODEOWNERS (hunk 4 above)
Comment (crazygao) [reply .github/CODEOWNERS:36]: OK，merge these。

File: .github/CODEOWNERS (hunk 2 above)
Comment (crazygao) [reply .github/CODEOWNERS:12]: OKay.

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -1,7 +1,29 @@
-* @microsoft/prompt-flow-approvers
+# CAUTION: Order is important; the LAST matching pattern takes the most precedence.
+# So put outside folders first, and then inside folders.
+# Don't change the sequence of the regions.
 
-/src/promptflow-tracing/promptflow/tracing/_start_trace.py @microsoft/promptflow-sdk
+* @microsoft/prompt-flow-approver
```
Comment (wangchao1230) [inline .github/CODEOWNERS:5]: This is not correct

File: .github/CODEOWNERS (hunk 1 above)
Comment (crazygao) [reply .github/CODEOWNERS:16]: GPT auto generation....

File: .github/CODEOWNERS (hunk 5 above)
Comment (crazygao) [reply .github/CODEOWNERS:5]: Yes! Great.

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**